sphinxcontrib/needs/directives/needtable.py

```diff
@@ -154,13 +166,19 @@ def process_needtables(app, doctree, fromdocname):
         tgroup = nodes.tgroup()
 
         # Define Table column width
-        # ToDo: Find a way to chosen to perfect width automatically.
+        colwidths_counter = 0
+        colwidths = current_needtable["colwidths"]
         for option, title in current_needtable["columns"]:
-            if option == "TITLE":
+
+            if colwidths:  # Get values from given colwidths option
+                tgroup += nodes.colspec(colwidth=int(colwidths[colwidths_counter]))
+            elif option == "TITLE":  # if nothing in colwidths...
                 tgroup += nodes.colspec(colwidth=15)
             else:
                 tgroup += nodes.colspec(colwidth=5)
 
+            colwidths_counter += 1
+
```

Comment:
`option` and `title` don't appear to be used in the loop. why not iterate over the colwidths directly?

Reply:
Option is used in the if-clause.
But yes, this could overall be restructured. 
But it's working, so I keep it :)